Extensions/xkit_patches.js

```diff
@@ -223,6 +223,14 @@ XKit.tools.dump_config = function(){
 	}, {});
 };
 
+// http://stackoverflow.com/a/901144/2073440
+XKit.tools.getParameterByName = function(name){
+	name = encodeURIComponent(name);
+	var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
+		results = regex.exec(location.search);
+	return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
```

Comment:
we're decoding URI component twice here.

also, a nit, but I think the ternary, function call, array index, and replace method is a little too much complexity for one line and would prefer to see it separated.